chain/stmgr/utils.go

```diff
@@ -249,35 +227,30 @@ func GetSectorsForWinningPoSt(ctx context.Context, pv ffiwrapper.Verifier, sm *S
 		return nil, xerrors.Errorf("generating winning post challenges: %w", err)
 	}
 
-	// we don't need to filter here (and it's **very** slow)
-	sectors, err := mas.LoadSectorsFromSet(nil, false)
+	iter, err := provingSectors.BitIterator()
 	if err != nil {
-		return nil, xerrors.Errorf("loading proving sectors: %w", err)
+		return nil, xerrors.Errorf("iterating over proving sectors: %w", err)
 	}
 
-	out := make([]proof.SectorInfo, len(ids))
-	for i, n := range ids {
-		sb, err := provingSectors.Slice(n, 1)
-		if err != nil {
-			return nil, err
-		}
-
-		sid, err := sb.First()
-		if err != nil {
-			return nil, err
-		}
-
-		var sinfo miner.SectorOnChainInfo
-		found, err := sectors.Get(sid, &sinfo)
-
+	// Select winning sectors by _index_ in the all-sectors bitfield.
+	selectedSectors := bitfield.New()
+	prev := uint64(0)
+	for _, n := range ids {
+		sno, err := iter.Nth(n - prev)
 		if err != nil {
-			return nil, xerrors.Errorf("loading sector info: %w", err)
+			return nil, xerrors.Errorf("iterating over proving sectors: %w", err)
 		}
+		selectedSectors.Set(sno)
+		prev = n
+	}
```

Comment:
@magik6k could you carefully review this code? I think it's right, but I'm not sure if it's tested anywhere.

Reply:
Looks good